.github/workflows/main.yaml

```diff
@@ -0,0 +1,160 @@
+            --wait --log
+      - run: echo "🍏 This job's status is ${{ job.status }}."              
+  development:
+    needs: publish
+    runs-on: self-hosted
+    steps:
+      - name: Setup Runner for Argo
+        run: |
+            cd $HOME
+            echo "Install argo"
+            # Download the binary
+            curl -sLO https://github.com/argoproj/argo-workflows/releases/download/${ARGO_VERSION}/argo-linux-amd64.gz
+            # Unzip
+            gunzip argo-linux-amd64.gz
+            # Make binary executable
+            chmod +x argo-linux-amd64
+            # Move binary to path
+            sudo mv ./argo-linux-amd64 /usr/local/bin/argo
+            # Test installation
+            argo version --short            
+      - name: Check out repository code
+        uses: actions/checkout@v3            
+      - name: development
+        run: |
+          echo "commit sha $GITHUB_SHA"
+          argo version --short   
+          argo submit .argo/deploy.yaml \
+            --generate-name="${CI_PROJECT_NAME}-development-${GITHUB_SHA:7}" \
+            --serviceaccount=argo \
+            -p appName="${CI_PROJECT_NAME}" \
+            -p ciCommitSha="${GITHUB_SHA}" \
+            -p branch="${GITHUB_REF_NAME}" \
+            -p environment="development" \
+            -p gitRepoUrl="${GITREPO_URL}" \
+            -p shortSha="${GITHUB_SHA:7}" \
+            --wait --log  
+      - run: echo "🍏 This job's status is ${{ job.status }}."              
+  staging:
+    needs: development
+    runs-on: self-hosted
+    steps:
+      - name: Setup Runner for Argo
+        run: |
+            cd $HOME
+            echo "Install argo"
+            # Download the binary
+            curl -sLO https://github.com/argoproj/argo-workflows/releases/download/${ARGO_VERSION}/argo-linux-amd64.gz
+            # Unzip
+            gunzip argo-linux-amd64.gz
+            # Make binary executable
+            chmod +x argo-linux-amd64
+            # Move binary to path
+            sudo mv ./argo-linux-amd64 /usr/local/bin/argo
+            # Test installation
+            argo version --short            
+      - name: Check out repository code
+        uses: actions/checkout@v3              
+      - name: staging
+        run: |
+          echo "commit sha $GITHUB_SHA"
+          argo version --short   
+          argo submit .argo/deploy.yaml \
+            --generate-name="${CI_PROJECT_NAME}-staging-${GITHUB_SHA:7}" \
+            --serviceaccount=argo \
+            -p appName="${CI_PROJECT_NAME}" \
+            -p ciCommitSha="${GITHUB_SHA}" \
+            -p branch="${GITHUB_REF_NAME}" \
+            -p environment="staging" \
+            -p gitRepoUrl="${GITREPO_URL}" \
+            -p shortSha="${GITHUB_SHA:7}" \
+            --wait --log   
+      - run: echo "🍏 This job's status is ${{ job.status }}."              
+  release:
+    needs: staging  
+    runs-on: self-hosted
+    steps:
+      - name: Setup Runner for Argo
```

Comment:
This Step will in the future become part of runner:https://github.com/kubefirst/kubefirst/issues/397